glean-core/rlb/src/core_metrics.rs

```diff
@@ -23,76 +26,90 @@ impl ClientInfoMetrics {
     }
 }
 
-#[derive(Debug)]
-pub struct InternalMetrics {
-    pub app_build: StringMetric,
-    pub app_display_version: StringMetric,
-    pub app_channel: StringMetric,
-    pub os_version: StringMetric,
-    pub architecture: StringMetric,
-    pub device_manufacturer: StringMetric,
-    pub device_model: StringMetric,
-}
+pub mod internal_metrics {
+    use super::*;
 
-impl InternalMetrics {
-    pub fn new() -> Self {
-        Self {
-            app_build: StringMetric::new(CommonMetricData {
-                name: "app_build".into(),
-                category: "".into(),
-                send_in_pings: vec!["glean_client_info".into()],
-                lifetime: Lifetime::Application,
-                disabled: false,
-                dynamic_label: None,
-            }),
-            app_display_version: StringMetric::new(CommonMetricData {
-                name: "app_display_version".into(),
-                category: "".into(),
-                send_in_pings: vec!["glean_client_info".into()],
-                lifetime: Lifetime::Application,
-                disabled: false,
-                dynamic_label: None,
-            }),
-            app_channel: StringMetric::new(CommonMetricData {
-                name: "app_channel".into(),
-                category: "".into(),
-                send_in_pings: vec!["glean_client_info".into()],
-                lifetime: Lifetime::Application,
-                disabled: false,
-                dynamic_label: None,
-            }),
-            os_version: StringMetric::new(CommonMetricData {
-                name: "os_version".into(),
-                category: "".into(),
-                send_in_pings: vec!["glean_client_info".into()],
-                lifetime: Lifetime::Application,
-                disabled: false,
-                dynamic_label: None,
-            }),
-            architecture: StringMetric::new(CommonMetricData {
-                name: "architecture".into(),
-                category: "".into(),
-                send_in_pings: vec!["glean_client_info".into()],
-                lifetime: Lifetime::Application,
-                disabled: false,
-                dynamic_label: None,
-            }),
-            device_manufacturer: StringMetric::new(CommonMetricData {
-                name: "device_manufacturer".into(),
-                category: "".into(),
-                send_in_pings: vec!["glean_client_info".into()],
-                lifetime: Lifetime::Application,
-                disabled: false,
-                dynamic_label: None,
-            }),
-            device_model: StringMetric::new(CommonMetricData {
-                name: "device_model".into(),
-                category: "".into(),
-                send_in_pings: vec!["glean_client_info".into()],
-                lifetime: Lifetime::Application,
-                disabled: false,
-                dynamic_label: None,
-            }),
-        }
-    }
+    #[allow(non_upper_case_globals)]
```

Comment:
You could move this to the top of the file as:

```rust
#![allow(non_upper_case_globals)]
```

Reply:
Oh right. If it doesn't bother you too much, I'd save another change and do that as a follow-up: we're going to re-generate this file anyway!

Reply:
Sure